api-router.js

```diff
@@ -4,14 +4,38 @@ const calcTxInfo = require('./util/calc-tx-info.js');
 const rskTokenBridgeController = require('./rsk-token-bridge-controller.js');
 const getAddressReport = require('./util/address-report.js');
 const rskActivityReport = require('./util/activity-report/index.js');
+const {
+  verifyParams,
+  readCache,
+  updateCache,
+  getParamValues,
+} = require('./util/functions.js');
 
 const router = express.Router();
 
 router.use(function (req, res, next) {
-  console.log(Date.now());
+  console.log(new Date());
   next();
 });
 
+// universal cache handling middleware
+// returns cached data immediately after receiving a request
+// and then then tries to update cache from the DB
+router.use('/rsk-activity-report/avg-tx-cost', async (req, res) => {
```

Comment:
this is a universal middleware. for now it only works for `/rsk-activity-report/avg-tx-cost` but later I'll match all other routes

Reply:
LGTM